server/api/crud/alerts.py

```diff
@@ -0,0 +1,279 @@
+
+            active = False
+            if send_notification:
+                state.count += 1
+                logger.debug(f"Sending notifications for alert {alert.name}")
+                AlertNotificationPusher().push(alert, event_data)
+
+                if alert.reset_policy == "auto":
+                    self.reset_alert(session, alert.project, alert.name)
+                else:
+                    active = True
+
+            server.api.utils.singletons.db.get_db().store_alert_state(
+                session,
+                alert.project,
+                alert.name,
+                count=state.count,
+                last_updated=event_data.timestamp,
+                obj=state_obj,
+                active=active,
+            )
+
+    @staticmethod
+    def _validate_alert(alert, name, project):
+        if name != alert.name:
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Alert name mismatch for alert {name} for project {project}. Provided {alert.name}"
+            )
+
+        if (
+            alert.criteria is not None
+            and alert.criteria.period is not None
+            and Alerts._string2datetime(alert.criteria.period) is None
+        ):
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Invalid period ({alert.criteria.period}) specified for alert {name} for project {project}"
+            )
+
+        for notification in alert.notifications:
+            if notification.kind not in [
+                mlrun.common.schemas.NotificationKind.git,
+                mlrun.common.schemas.NotificationKind.slack,
+                mlrun.common.schemas.NotificationKind.webhook,
+            ]:
+                raise mlrun.errors.MLRunBadRequestError(
+                    f"Unsupported notification ({notification.kind}) for alert {name} for project {project}"
+                )
+            notification_object = mlrun.model.Notification.from_dict(
+                notification.dict()
+            )
+            notification_object.validate_notification()
+
+        if alert.entity.project != project:
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Invalid alert entity project ({alert.entity.project}) for alert {name} for project {project}"
+            )
+
+    @staticmethod
+    def _string2datetime(date_str):
+        date_str = date_str.strip().lower()
+        s = 0
+        m = 0
+        h = 0
+        d = 0
+        if date_str.endswith("d"):
+            d = int(date_str.split("d")[0].strip())
+        elif date_str.endswith("h"):
+            h = int(date_str.split("h")[0].strip())
+        elif date_str.endswith("m"):
+            m = int(date_str.split("m")[0].strip())
+        elif date_str.endswith("s"):
+            s = int(date_str.split("s")[0].strip())
+
+        if s == 0 and m == 0 and h == 0 and d == 0:
+            return None
+
+        return datetime.timedelta(days=d, hours=h, minutes=m, seconds=s)
```

Comment:
Use `humanfriendly.parse_timespan` to convert the string to seconds, so something like:
```
datetime.timedelta(seconds=humanfriendly.parse_timespan(date_str))